amulet/level/formats/sponge_schem/format_wrapper.py

```diff
@@ -97,191 +151,117 @@ def _create(
-                        ]
-                    )
-                    cx, cz = numpy.floor([x, z]).astype(int) >> 4
-                    if (cx, cz) in self._chunks and (x, y, z) in self._chunks[
-                        (cx, cz)
-                    ].selection:
-                        self._chunks[(cx, cz)].entities.append(entity)
-
+            palette_tag = blk_ctn.get("BlockPalette") or blk_ctn.get("Palette")
+            data_tag = blk_ctn.get("Data")
+            block_entities_tag = blk_ctn.get("BlockEntities")
+
+        if not isinstance(palette_tag, CompoundTag):
+            raise SpongeSchemReadError("Missing or invalid palette tag")
+        if not isinstance(data_tag, (ByteArrayTag, IntArrayTag)):
+            raise SpongeSchemReadError("Missing or invalid 'Data' tag")
+
+        block_palette = {
+            idx.py_int: Block.from_string_blockstate(state)
+            for state, idx in palette_tag.items()
+        }
+
+        if isinstance(data_tag, IntArrayTag):
+            unpacked = data_tag.np_array.tolist()
         else:
-            raise SpongeSchemReadError(
-                f"Sponge Schematic Version {version} is not supported currently."
-            )
+            raw = numpy.array(data_tag, dtype=numpy.uint8)
+            unpacked = decode_byte_array(raw)
+
+        expected = size[0] * size[1] * size[2]
+        if len(unpacked) != expected:
+            raise SpongeSchemReadError("Block data length does not match dimensions")
+
+        dy, dz, dx = size[1], size[2], size[0]
+        blocks_array = (
+            numpy.array(unpacked, dtype=numpy.uint32)
+            .reshape((dy, dz, dx))
+            .transpose((2, 0, 1))
+        )
+
+        for cx, cz in selection.chunk_locations():
+            cbox = SelectionBox.create_chunk_box(cx, cz).intersection(selection)
+            sl = cbox.create_moved_box(selection.min, subtract=True).slice
+            sub = blocks_array[sl]
+            ids, remap = numpy.unique(sub, return_inverse=True)
+            remap = remap.reshape(sub.shape)
+
+            pal = numpy.empty(len(ids), dtype=object)
+            for i, orig in enumerate(ids):
+                pal[i] = block_palette[orig]
+
+            self._chunks[(cx, cz)] = SpongeSchemChunk(cbox, remap, pal, [], [])
+
+        if isinstance(block_entities_tag, ListTag):
+            for be in block_entities_tag:
+                pos = be.get("Pos")
+                if isinstance(pos, IntArrayTag) and len(pos) == 3:
+                    world_pos = pos.np_array + min_point
+                    x, y, z = world_pos
+                    be["Pos"] = IntArrayTag(world_pos)
+                    key = (x >> 4, z >> 4)
+                    if key in self._chunks and (x, y, z) in self._chunks[key].selection:
+                        self._chunks[key].block_entities.append(be)
+
+        entities = sponge_schem.get("Entities")
+        if isinstance(entities, ListTag):
+            for ent in entities:
```

Comment:
You aren't validating the type of ent

Reply:
I think I accidentally deleted some stuff. I'm going to go over and use the original code as a base and copy my stuff into it, that way, it should be clearer what was changed and I can verify nothing else was removed.

Reply:
Yeah, I removed that line accidentally. Just double-checked to make sure nothing else was removed.  I'll work on some clean up later.